packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts

```diff
@@ -86,6 +88,26 @@ export class FargateService extends BaseService implements IFargateService {
       throw new Error('A TaskDefinition must have at least one essential container');
     }
   }
+
+  /**
+   * Adds AWS X-Ray daemon as a sidecar container to enable tracing
+   */
+  public addTracing(props: TracingOptions = {}) {
+    // TODO: adjust task size based on container-level resources (cpu/memory)?
+    const optIn = props.enableLogging !== undefined ? props.enableLogging : true;
```

Comment:
Like this could be reused between fargate and ec2, even through an internal utility function

Reply:
Addressed